src/libraries/Common/src/System/Security/Cryptography/EccKeyFormatHelper.cs

```diff
@@ -269,23 +269,12 @@ private static ECCurve GetCurve(ECDomainParameters domainParameters)
                 throw new CryptographicException(SR.Cryptography_ECC_NamedCurvesOnly);
             }
 
-            Oid curveOid;
-
-            switch (domainParameters.Named)
-            {
-                case Oids.secp256r1:
-                    curveOid = new Oid(Oids.secp256r1, nameof(ECCurve.NamedCurves.nistP256));
-                    break;
-                case Oids.secp384r1:
-                    curveOid = new Oid(Oids.secp384r1, nameof(ECCurve.NamedCurves.nistP384));
-                    break;
-                case Oids.secp521r1:
-                    curveOid = new Oid(Oids.secp521r1, nameof(ECCurve.NamedCurves.nistP521));
-                    break;
-                default:
-                    curveOid = new Oid(domainParameters.Named);
-                    break;
-            }
+            Oid curveOid = domainParameters.Named switch {
+                Oids.secp256r1 => Oids.secp256r1Oid,
+                Oids.secp384r1 => Oids.secp384r1Oid,
+                Oids.secp521r1 => Oids.secp521r1Oid,
+                _ => new Oid(domainParameters.Named)
```

Comment:
May as well fix this to use the cheaper ctor while we're here.

```suggestion
                _ => new Oid(domainParameters.Named, null)
```